src/twisted/web/server.py

```diff
@@ -525,7 +525,8 @@ def encoderForRequest(self, request):
         """
         acceptHeaders = request.requestHeaders.getRawHeaders(
             'accept-encoding', [])
-        supported = ','.join(acceptHeaders).split(',')
+        supported = [encoding.strip()
+                     for encoding in ','.join(acceptHeaders).split(',')]
```

Comment:
I don't think that regexp is quite right:

```py
>>> import re
>>> gzipRe = re.compile(br'[\s,]?gzip[\s,]?')
>>> gzipRe.match(b'gzip')
<_sre.SRE_Match object; span=(0, 4), match=b'gzip'>
>>> gzipRe.match(b'x-gzip')
>>> gzipRe.match(b'gzipzzz')  # should not match
<_sre.SRE_Match object; span=(0, 4), match=b'gzip'>
>>> bool(gzipRe.match(b'deflate, gzip'))  # match() is not right
False
```

The regexp should use `^` and `$` to allow matches at the beginning and end of the string. `search()` should be used to allow non-prefix matches:

```py
>>> gzipRe2 = re.compile(br'(:?^|[\s,])gzip(:?$|[\s,])')
>>> bool(gzipRe2.search(b'gzip'))
True
>>> bool(gzipRe2.search(b'deflate, gzip'))
True
>>> bool(gzipRe2.search(b'deflate, gzip,gzip'))
True
>>> bool(gzipRe2.search(b'deflate, gzip,br'))
True
>>> bool(gzipRe2.search(b'deflate,gzip,br'))
True
>>> bool(gzipRe2.search(b'x-gzip,br'))
False
>>> bool(gzipRe2.search(b'deflate,br'))
False
```

I would keep everything as bytes so that we don't need to decode the header value.